### PR TITLE
Properly save liquid type and location when bucket-fill

### PR DESCRIPTION
**Pull request by rumickon**
slab
Poured and filled water into slab block.

Before PR:
before

After PR:
after

### CODE REVIEW

**Review by alchemistmatt — DISMISSED** [review]
Looks good to me: if it's a waterlogged block, we should record that the bucket was filled with water, and not ~lava~ milk.  For reference, search for `Waterlogged` at https://papermc.io/javadocs/paper/1.15/org/bukkit/Material.html

**Comment by addstar** [comment]
@rumickon Can you update the PR description to explain what your changes do and what problem it solves?

I believe @alchemistmatt already summarised it, but as the author, I think it's up to you to provide a clear change description (changes, what it fixes, background info, etc). It's always important for us to know if it's been tested and how. Thanks :)

File: src/main/java/me/botsko/prism/listeners/PrismPlayerEvents.java [hunk 1]
```diff
@@ -282,13 +282,14 @@ public void onPlayerBucketFill(final PlayerBucketFillEvent event) {
 		final Player player = event.getPlayer();
 		if (!Prism.getIgnore().event("bucket-fill", player))
 			return;
-		final Block spot = event.getBlockClicked().getRelative(event.getBlockFace());
+		final Block spot = event.getBlock();
```
Comment (rumickon) [inline src/main/java/me/botsko/prism/listeners/PrismPlayerEvents.java]: Properly save location of block.

File: src/main/java/me/botsko/prism/listeners/PrismPlayerEvents.java [hunk 2]
```diff
@@ -282,13 +282,14 @@ public void onPlayerBucketFill(final PlayerBucketFillEvent event) {
...
 
 		String liquid_type = "milk";
 		if (spot.getType() == Material.WATER) {
 			liquid_type = "water";
-		}
-		else if (spot.getType() == Material.LAVA) {
+		} else if (spot.getBlockData() instanceof Waterlogged && ((Waterlogged) spot.getBlockData()).isWaterlogged()) {
```
Comment (rumickon) [inline src/main/java/me/botsko/prism/listeners/PrismPlayerEvents.java]: Tracing waterlogged blocks.

**Comment by rumickon** [comment]
I tried to explain the changes. But I can’t test the latest merged commits, because Prism does not process commands due to errors.

**Comment by Narimm** [comment]
The command issue is fixed I just need an approving review.

**Review by alchemistmatt — APPROVED** [review]
This change checks for a Waterlogged block; looks good

**Pull request merged**